[inputstream] change to new addon to kodi callback way

AlwinEsch opened the pull request:
This is the first major change to become multiple instances to work.

There are the callbacks moved to the ADDON::CInputStream class. Reason is
to have this parts together and later with multiple instances to call
always right related ADDON::CInputStream class.

On addon is currently nothing needed to change, only a recompile to
match them.

<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

hauihau commented on xbmc/addons/interfaces/AddonInterfaces.h:
```diff
@@ -73,7 +73,6 @@ namespace ADDON
     \*/
     static void*        INPUTSTREAMLib_RegisterMe      (void *addonData);
     static void         INPUTSTREAMLib_UnRegisterMe    (void *addonData, void* cbTable);
-    void*               GetHelperInputStream()        { return m_helperInputStream; }
```
did you forget the m_helperInputStream some lines below?

AlwinEsch replied:
Oh, yes. Thanks a lot!

AlwinEsch commented:
jenkins build this please

AlwinEsch committed:
[inputstream] change to new addon to kodi callback way

This is the first major change to become multiple instances to work.

There are the callbacks moved to the ADDON::CInputStream class. Reason is
to have this parts together and later with multiple instances to call
always right related ADDON::CInputStream class.

On addon is currently nothing needed to change, only a recompile to
match them.

AlwinEsch commented:
jenkins build this please

ksooo reviewed (approved):
Looking good.
